src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -44,12 +142,16 @@ public virtual int PrefetchCount
                 {
                     throw Fx.Exception.ArgumentOutOfRange(nameof(this.PrefetchCount), value, "Value must be greater than 0");
                 }
-
                 this.prefetchCount = value;
+                ReceivingAmqpLink link;
+                if (this.ReceiveLinkManager.TryGetOpenedObject(out link))
+                {
+                    link.SetTotalLinkCredit((uint)value, true, true);
+                }
             }
         }
 
-        public virtual long LastPeekedSequenceNumber
+        public long LastPeekedSequenceNumber
```

Comment:
We define this in the `IMessageReceiver` interface. Whats your thought for making it protected?